ReactAndroid/src/main/java/com/facebook/react/modules/datepicker/DatePickerDialogFragment.java

```diff
@@ -53,8 +54,44 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
     final int month = c.get(Calendar.MONTH);
     final int day = c.get(Calendar.DAY_OF_MONTH);
 
-    final DatePickerDialog dialog =
-        new DismissableDatePickerDialog(activityContext, onDateSetListener, year, month, day);
+    DatePickerMode mode = DatePickerMode.DEFAULT;  //To set mode to default
+    if (args != null && args.getString(DatePickerDialogModule.ARG_MODE, null) != null) {
+      mode = DatePickerMode.fromString(args.getString(DatePickerDialogModule.ARG_MODE));
+    }
+
+    DatePickerDialog dialog = null;
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) { // To support material calendar date-picker
+      switch (mode) {
+        case CALENDAR:
+          dialog = new DismissableDatePickerDialog(activityContext,
+            activityContext.getResources().getIdentifier("CalendarDatePickerDialog", "style", activityContext.getPackageName()),
+            onDateSetListener, year, month, day);
+          break;
+        case SPINNER:
+          dialog = new DismissableDatePickerDialog(activityContext,
+            activityContext.getResources().getIdentifier("SpinnerDatePickerDialog", "style", activityContext.getPackageName()),
+            onDateSetListener, year, month, day);
+          break;
+        case DEFAULT:
+          //To Show a default native date picker
+          dialog = new DismissableDatePickerDialog(activityContext, onDateSetListener, year, month, day);
+          break;
+      }
+    } else {
+      dialog = new DismissableDatePickerDialog(activityContext, onDateSetListener, year, month, day);
+
+      switch (mode) {
+        case CALENDAR:
+          dialog.getDatePicker().setCalendarViewShown(true);
+          dialog.getDatePicker().setSpinnersShown(false);
+          break;
+        case SPINNER:
+          dialog.getDatePicker().setCalendarViewShown(false);
+          break;
```

Comment:
The code in this switch statement should be moved up to the branches where the dialogs are created.